core/domain/app_feedback_report_services.py

```diff
@@ -0,0 +1,650 @@
+                entry_point_name: 1
+            },
+            constants.STATS_PARAMETER_NAMES.text_language_code.name: {
+                text_language_code: 1
+            },
+            constants.STATS_PARAMETER_NAMES.audio_language_code.name: {
+                audio_language_code: 1
+            },
+            constants.STATS_PARAMETER_NAMES.android_sdk_version.name: {
+                sdk_version: 1
+            },
+            constants.STATS_PARAMETER_NAMES.version_name.name: {
+                version_name: 1
+            }
+        }
+        app_feedback_report_models.AppFeedbackReportStatsModel.create(
+            stats_id, platform, ticket_id, date, 0, stats_dict)
+        stats_model = (
+            app_feedback_report_models.AppFeedbackReportStatsModel.get_by_id(
+                stats_id))
+    else:
+        # Update existing stats model.
+        stats_dict = stats_model.daily_param_stats
+
+        stats_dict[constants.STATS_PARAMETER_NAMES.report_type.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[constants.STATS_PARAMETER_NAMES.report_type.name],
+                report_type, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.country_locale_code.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[
+                    constants.STATS_PARAMETER_NAMES.country_locale_code.name],
+                country_locale_code, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.entry_point_name.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[
+                    constants.STATS_PARAMETER_NAMES.entry_point_name.name],
+                entry_point_name, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.audio_language_code.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[
+                    constants.STATS_PARAMETER_NAMES.audio_language_code.name],
+                audio_language_code, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.text_language_code.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[
+                    constants.STATS_PARAMETER_NAMES.text_language_code.name],
+                text_language_code, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.android_sdk_version.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[
+                    constants.STATS_PARAMETER_NAMES.android_sdk_version.name],
+                sdk_version, delta))
+        stats_dict[constants.STATS_PARAMETER_NAMES.version_name.name] = (
+            calculate_new_stats_count_for_parameter(
+                stats_dict[constants.STATS_PARAMETER_NAMES.version_name.name],
+                version_name, delta))
+
+    stats_model.daily_param_stats = stats_dict
+    stats_model.total_reports_submitted += delta
```

Comment:
Will this overcount for the initial case? It seems like it'll always be one off since we initialize to 1, then add the delta.

Reply:
I don't believe so -- this is the only place where we set this value when updating the stats, we're only updating individual field stats in the above if / else. We initialize the total_reports_submitted to 0 when creating the stats model (L155).

Reply:
Ah I see. I misunderstood--thanks for clarifying.